tests/Unit/ApparentHorizons/Test_ApparentHorizonFinder.cpp

```diff
@@ -0,0 +1,350 @@
+        get<StrahlkorperTags::Strahlkorper<Frame::Inertial>>(box);
+    // We don't choose many grid points (for speed of test), so we
+    // don't get the horizon radius extremely accurately.
+    Approx custom_approx = Approx::custom().epsilon(1.e-2).scale(1.0);
+    CHECK(strahlkorper.average_radius() == custom_approx(2.0));
+    ++test_schwarzschild_horizon_called;
+  }
+};
+
+// Counter to ensure that this function is called
+size_t test_kerr_horizon_called = 0;
+struct TestKerrHorizon {
+  template <typename DbTags, typename Metavariables>
+  static void apply(
+      const db::DataBox<DbTags>& box,
+      const Parallel::ConstGlobalCache<Metavariables>& /*cache*/,
+      const typename Metavariables::temporal_id& /*temporal_id*/) noexcept {
+    const auto& strahlkorper =
+        get<StrahlkorperTags::Strahlkorper<Frame::Inertial>>(box);
+    // Test actual horizon radius against analytic value at the same
+    // theta,phi points.
+    const auto expected_radius = gr::Solutions::kerr_horizon_radius(
+        strahlkorper.ylm_spherepack().theta_phi_points(), 1.1,
+        {{0.12, 0.23, 0.45}});
+    const auto& horizon_radius =
+        get<StrahlkorperTags::Radius<Frame::Inertial>>(box);
+    // The accuracy is not great because I use only a few grid points
+    // to speed up the test.
+    Approx custom_approx = Approx::custom().epsilon(1.e-3).scale(1.0);
+    CHECK_ITERABLE_CUSTOM_APPROX(horizon_radius, get(expected_radius),
+                                 custom_approx);
+    ++test_kerr_horizon_called;
+  }
+};
+
+template <typename Metavariables, typename InterpolationTargetTag>
+struct mock_interpolation_target {
+  using metavariables = Metavariables;
+  using chare_type = ActionTesting::MockArrayChare;
+  using array_index = size_t;
+  using action_list = tmpl::list<>;
+  using component_being_mocked =
+      intrp::InterpolationTarget<Metavariables, InterpolationTargetTag>;
+  using initial_databox = db::compute_databox_type<
+      typename intrp::Actions::InitializeInterpolationTarget<
+          InterpolationTargetTag>::template return_tag_list<Metavariables>>;
+
+  using const_global_cache_tag_list = Parallel::get_const_global_cache_tags<
+      tmpl::list<typename InterpolationTargetTag::compute_target_points,
+                 typename InterpolationTargetTag::post_interpolation_callback>>;
+};
+
+template <typename Metavariables>
+struct mock_interpolator {
+  using metavariables = Metavariables;
+  using chare_type = ActionTesting::MockArrayChare;
+  using array_index = size_t;
+  using const_global_cache_tag_list = tmpl::list<>;
+  using action_list = tmpl::list<>;
+  using component_being_mocked = intrp::Interpolator<Metavariables>;
+  using initial_databox =
+      db::compute_databox_type<typename intrp::Actions::InitializeInterpolator::
+                                   template return_tag_list<Metavariables>>;
+};
+
+template <typename PostHorizonFindCallback>
+struct MockMetavariables {
+  struct AhA {
```

Comment:
Technically it could be moved out, but it is conceptually part of Metavariables.  And everywhere else something like this is done, it is inside Metavariables, so I like to keep consistency.